driverdataaggregation/README.md

```diff
@@ -0,0 +1,407 @@
+# Data Aggregation Scaling Tests
+
+SmartSim-Scaling currently offers three data aggregation test versions:
+
+ 1. Data Aggregation Standard               (C++ Client and SmartRedis Orchestrator)
+ 2. Data Aggregation Standard Python        (Python Client and SmartRedis Orchestrator)
+ 3. Data Aggregation Standard File System   (Python Client and File System)
+
+## Client Description
+
+The data aggregation scaling test runs two applications. The first application
+is an MPI application that produces datasets that are added to an aggregation list.
+In this producer application, each MPI rank has a single-threaded client.  The second
+application is a consumer application. This application consumes the aggregation
+lists that are produced by the first application.  The consumer application
+can be configured to use multiple threads for data aggregation.  The producer and consumer
+applications are running at the same time, but the producer application waits for the
+consumer application to finish an aggregation list before starting to produce
+the next aggregation list.
+
+By default, the clients in the producer application perform an adjustable 100 executions of the following command:
+
+  1) ``append_to_list`` (add dataset to the aggregation list)
+
+Note that the client on rank 0 of the producer application performs a ``get_list_length()``
+function invocation prior to an ``MPI_BARRIER`` in order to only produce the next aggregation
+list after the previous aggregation list was consumed by the consumer application.
+
+There is only a single MPI rank for the consumer application, which means there is only
+one SmartRedis client active for the consumer application.  The consumer application client
```

Comment:
This seems very strange, please confirm

Reply:
Did we just not want to deal with the synchronization issues here?

Reply:
the producer is a bunch of individual tasks that produce data and the consumer aggregates all the lists individually in serial